### PR TITLE
Error if the user is trying to implement DO's in a service worker

### PR DESCRIPTION
**Pull request by caass**
I think...this...is good..?

Continuation of https://github.com/cloudflare/wrangler2/pull/634

Closes https://github.com/cloudflare/wrangler2/issues/635

### CODE REVIEW

File: packages/wrangler/src/entry.ts [hunk 1]
```diff
@@ -48,6 +48,22 @@ export async function getEntry(
     directory,
     args.format ?? config.build?.upload?.format
   );
+
+  if (format === "service-worker" && hasDurableObjectImplementations(config)) {
+    const errorMessage =
+      "You seem to be trying to use Durable Objects with Service Worker syntax.";
+    const addScriptName =
+      "You can use Durable Objects defined in other (Module) Workers from a Service Worker by specifying a `script_name` in your wrangler.toml,where `script_name` is the name of the worker that implements that Durable Object. For example:";
```
Comment (petebacondarwin) [inline packages/wrangler/src/entry.ts:56]:
```suggestion
      "You can use Durable Objects defined in other (Module) Workers from a Service Worker by specifying a `script_name` in your wrangler.toml, where `script_name` is the name of the worker that implements that Durable Object. For example:";
```

File: packages/wrangler/src/entry.ts [hunk 2]
```diff
@@ -48,6 +48,22 @@ export async function getEntry(
...
+    const addScriptNameExamples = generateAddScriptNameExamples(config);
+    const migrateText =
+      "Alternatively, migrate your worker to Module format to implement a new Durable Object:";
```
Comment (petebacondarwin) [inline packages/wrangler/src/entry.ts:59]: What is the convention? Should this be `ES Module format`?

File: packages/wrangler/src/entry.ts [hunk 3]
```diff
@@ -157,3 +173,48 @@ export function fileExists(filePath: string): boolean {
...
+ * config indicating how the user can add a `script_name` field to bind an
+ * externally defined Durable Object.
+ */
+function generateAddScriptNameExamples(config: Config): string {
+  const allBindings = [
+    ...config.durable_objects.bindings,
+    ...Object.values(config.env)
+      .map((env) => env.durable_objects.bindings)
+      .flat(),
+  ];
```
Comment (petebacondarwin) [inline packages/wrangler/src/entry.ts:204]: NIT: this could be pulled out into a shared function.

**Review by petebacondarwin — COMMENTED** [review]
Looking good to me.

File: packages/wrangler/src/entry.ts (hunk 2 above)
Comment (caass) [reply packages/wrangler/src/entry.ts:59]: [It's unclear](https://developers.cloudflare.com/workers/learning/migrating-to-module-workers/) to me :(

Comment (petebacondarwin) [reply packages/wrangler/src/entry.ts:59]: Either works for me.

**Review by threepointone — APPROVED** [review]
Thank you for the extensive tests! This looks great, let's ship it.

**Pull request merged**